control/pid_longitudinal_controller/src/longitudinal_controller_utils.cpp

```diff
@@ -129,11 +129,16 @@ double calcElevationAngle(const TrajectoryPoint & p_from, const TrajectoryPoint
 }
 
 Pose calcPoseAfterTimeDelay(
-  const Pose & current_pose, const double delay_time, const double current_vel)
+  const Pose & current_pose, const double delay_time, const double current_vel,
+  const double current_acc)
 {
+  if (delay_time <= 0.0 || current_vel <= 0.0) {
+    return current_pose;
+  }
   // simple linear prediction
   const double yaw = tf2::getYaw(current_pose.orientation);
-  const double running_distance = delay_time * current_vel;
+  const double running_distance =
+    delay_time * current_vel + 0.5 * current_acc * delay_time * delay_time;
```

Comment:
We'd better check if `running_distance` will not exceed 0.
For example, when the current_vel = 1, current_acc = -3, and delay_time = 1, `running_distance` will be negative which is not expected.

What do you think?

Reply:
Hi, thanks for advices, I updated the branch can you check it again?

Reply:
Thank you.
In the example I mentioned before when the current_vel = 1, current_acc = -3, and delay_time = 1, the `running_distance` will be a certain positive value, but the current implementation makes it zero.
Could you fix it just in case.

Reply:
`  const double running_distance =
    abs(delay_time * current_vel + 0.5 * current_acc * delay_time * delay_time);`
    
With the latest update, it will be 0.5, I couldn't understand your point.

Reply:
Ah, sorry I misunderstood your implementation.
With `abs`, the running distance will not be negative, but I'm not sure why we can use `abs` here.

What I wanted to say is that if applying the delay_time results in a negative running distance like the case where the current_vel = 1, current_acc = -3, and delay_time = 1, the time to reach zero velocity should be used to calculate running_distance.
The implementation will be as follows. Please check if the calculation is correct.
```cpp
const double running_distance = [&]() {
    const double time_to_zero_vel = current_vel / -current_acc;
    if (0 < time_to_zero_vel && time_to_zero_vel < delay_time) {
      return time_to_zero_vel * current_vel + 0.5 * current_acc * std::pow(time_to_zero_vel, 2);
    }
    return delay_time * current_vel + 0.5 * current_acc * std::pow(delay_time, 2);
}();
```

Reply:
Sorry for my mistake, you are right, why I don't know but I think that running distance always should be positive, it was the wrong idea. I updated the PR, can you check it again?